src/lib/routes/admin-api/feature-type.ts

```diff
@@ -50,15 +64,61 @@ export class FeatureTypeController extends Controller {
                 }),
             ],
         });
+
+        this.route({
+            method: 'put',
+            path: '/:id/lifetime',
+            handler: this.updateLifetime,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Unstable'],
+                    operationId: 'updateFeatureTypeLifetime',
+                    summary: 'Update feature type lifetime',
+                    description: `Updates the lifetime configuration for the specified [feature toggle type](https://docs.getunleash.io/reference/feature-toggle-types). The expected lifetime is an integer representing the number of days before Unleash marks a feature toggle of that type as potentially stale. If set to \`null\` or \`0\`, then feature toggles of that particular type will never be marked as potentially stale.
+
+When a feature toggle type's expected lifetime is changed, this will also cause any feature toggles of this type to be reevaluated for potential staleness.`,
+                    responses: {
+                        200: createResponseSchema('featureTypeSchema'),
+                        ...getStandardResponses(400, 401, 403, 404, 415),
+                    },
+                    requestBody: createRequestSchema(
+                        'updateFeatureTypeLifetimeSchema',
+                    ),
+                }),
+            ],
+        });
     }
 
     async getAllFeatureTypes(
         req: Request,
         res: Response<FeatureTypesSchema>,
     ): Promise<void> {
-        res.json({
-            version,
-            types: await this.featureTypeService.getAll(),
-        });
+        this.openApiService.respondWithValidation(
+            200,
+            res,
+            featureTypesSchema.$id,
+            {
+                version,
+                types: await this.featureTypeService.getAll(),
+            },
+        );
+    }
+
+    async updateLifetime(
+        req: IAuthRequest<
+            { id: string },
+            unknown,
+            UpdateFeatureTypeLifetimeSchema
+        >,
+        res: Response<FeatureTypeSchema>,
+    ): Promise<void> {
+        if (this.flagResolver.isEnabled('configurableFeatureTypeLifetimes')) {
+            throw new NotImplementedError(
+                "This operation isn't implemented yet",
+            );
+        } else {
+            res.status(409).end();
```

Comment:
`409` is not in responses. It's temporary, right?

Reply:
Good point! Yeah, it's only temporary until the endpoint gets stabilized. But good point; I'll add it to the endpoint description for now.